.github/workflows/nightly.yml

```diff
@@ -39,55 +44,137 @@ jobs:
+          - {os: macos-latest, target: x86_64-apple-darwin}
+        target_cpu: [generic, broadwell]
+        toolchain: [stable]
+        cross: [false]
+        include:
           # Windows
-          - {os: windows-latest, toolchain: stable-x86_64-pc-windows-gnu, target: x86_64-pc-windows-gnu, target_cpu: generic, cross: false}
-          - {os: windows-latest, toolchain: stable-x86_64-pc-windows-msvc, target: x86_64-pc-windows-msvc, target_cpu: generic, cross: false}
-          - {os: windows-latest, toolchain: stable-x86_64-pc-windows-gnu, target: x86_64-pc-windows-gnu, target_cpu: broadwell, cross: false}
-          - {os: windows-latest, toolchain: stable-x86_64-pc-windows-msvc, target: x86_64-pc-windows-msvc, target_cpu: broadwell, cross: false}
+          - config: {os: windows-latest, target: x86_64-pc-windows-gnu}
+            target_cpu: generic
+            toolchain: stable-x86_64-pc-windows-gnu
+            cross: false
+          - config: {os: windows-latest, target: x86_64-pc-windows-gnu}
+            target_cpu: broadwell
+            toolchain: stable-x86_64-pc-windows-gnu
+            cross: false
+          - config: {os: windows-latest, target: x86_64-pc-windows-msvc}
+            target_cpu: generic
+            toolchain: stable-x86_64-pc-windows-msvc
+            cross: false
+          - config: {os: windows-latest, target: x86_64-pc-windows-msvc}
+            target_cpu: broadwell
+            toolchain: stable-x86_64-pc-windows-msvc
+            cross: false
+          # Cross targets
+          - config: {os: ubuntu-latest, target: aarch64-unknown-linux-gnu}
+            target_cpu: generic
+            toolchain: stable
+            cross: true
+          - config: {os: ubuntu-latest, target: arm-unknown-linux-gnueabi}
+            target_cpu: generic
+            toolchain: stable
+            cross: true
+          - config: {os: ubuntu-latest, target: armv7-unknown-linux-gnueabihf}
+            target_cpu: generic
+            toolchain: stable
+            cross: true
+          - config: {os: ubuntu-latest,  target: x86_64-unknown-linux-musl}
+            target_cpu: generic
+            toolchain: stable
+            cross: true
+          - config: {os: ubuntu-latest,  target: x86_64-unknown-linux-musl}
+            target_cpu: broadwell
+            toolchain: stable
+            cross: true
+          - config: {os: ubuntu-latest, target: aarch64-linux-android}
+            target_cpu: generic
+            toolchain: stable
+            cross: true
+          # - config: {os: ubuntu-latest, target: mips64el-unknown-linux-gnuabi64}
+          #   target_cpu: generic
+          #   toolchain: stable
+          #   cross: true
+          # - config: {os: ubuntu-latest, target: powerpc64le-unknown-linux-gnu}
+          #   target_cpu: generic
+          #   toolchain: stable
+          #   cross: true
+
```

Comment:
I do not see the improvement here. With this change we are using a matrix but then specifying for each (or some) config pair extra features. I really think that by using the old style is clearer what configuration is run for each item even if it is redundant on some of its attributes.

Reply:
In fact we end up with the same redundant attributes:
```
 - config: {os: ubuntu-latest,  target: x86_64-unknown-linux-musl}
    target_cpu: generic
    toolchain: stable
    cross: true
```

Reply:
I wanted to get away from the giant horizontally extended table where omissions are easy to make, like the one that necessitated #2306.

Now, each entry has the same four attributes in a vertical layout that is easy to review, and the main matrix is used for its intended purpose: create an M x N set product of targets and CPUs.
Unfortunately, it's not possible to do more neat combinatorics with the current `matrix` features,
so the Windows build matrix had to be unrolled in the `include` list.